### ISSUE TITLE
gRPC server TRANSIENT_FAILURE after a certain time point

### ISSUE THREAD

**Issue by cmd23333** [grpc/grpc#32756]
### What version of gRPC and what language are you using?
Python 3.8.10
grpcio==1.43.0 (client)
grpcio==1.51.3 (server)

### What operating system (Linux, Windows,...) and version?
Ubuntu 20.04

### What runtime / compiler are you using (e.g. python version or version of gcc)
Python 3.8.10

### What did you do?
I start a grpc server in a docker, which has grpc version 1.51.3, and using UDS address.
In another docker, which has grpc version 1.43.0, I try to make a rpc call to server.

### What did you expect to see?
normally call and get result.

### What did you see instead?
failed to connect to all address.

```bash
[{"created":"@1631934123.441737107","description":"failed to connect to all addresses","file":"src/core/lib/transport/error_utils.cc","file_line":146,"grpc_status":14
```

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

### Anything else we should know about your project / environment?
This problem is not always appear, I mean, maybe in first few hours or days, the connection between the 2 docker is good enough. But after a certain time, I can't connect. Always can't. At this time, if I try to channel.subscribe(), the server's status will be TRANSIENT_FAILURE. If I restart the server docker, all is well.

This problem make our product unstable, is that because the version of grpc disagree? But, The 2 docker CAN connect before a certain point. What will make a server TRANSIENT_FAILURE?

**Comment by XuanWang-Amos** [comment]
I don't think it's related to version mismatch, TRANSIENT_FAILURE normally happens [because of connection issues](https://grpc.github.io/grpc/core/md_doc_connectivity-semantics-and-api.html), but it's hard to tell what's the issue here from the error itself.

We'll need more information in order to help further, you can get more logs by adding those env flags:
`GRPC_VERBOSITY=debug GRPC_TRACE=all,-timer,-timer_check`

**Comment by cmd23333** [comment]
> I don't think it's related to version mismatch, TRANSIENT_FAILURE normally happens [because of connection issues](https://grpc.github.io/grpc/core/md_doc_connectivity-semantics-and-api.html), but it's hard to tell what's the issue here from the error itself.
> 
> We'll need more information in order to help further, you can get more logs by adding those env flags: `GRPC_VERBOSITY=debug GRPC_TRACE=all,-timer,-timer_check`

Can I set these env flags AFTER the application is running? Because such problem is happened with a possibility, and we are deploying our product, can't always run with debug mode.

**Comment by gnossen** [comment]
> Can I set these env flags AFTER the application is running?

Unfortunately, no. If you're concerned about the volume of logging though, you can probably change the `GRPC_TRACE` variable to just `channel,server_channel,api`.

The fact that you're using a UDS rules out network connection issues. Is there anything you can do to help determine whether your server application has crashed or frozen?

**Comment by evgmoskalenko** [comment]
Maybe you have the same problem as me - https://github.com/grpc/grpc/issues/32435. And it happens sporadically after switching the network. Try doing nothing and see how quickly the connection is restored on your own.

In my case, it takes approximately 15-20 minutes.

In my case, we see network switching between regions on the metrics. Also, I created on Python service which sends ping-pong GRPC requests (between client and servers) and stores each request (latency and error rate) to the Prometheus metrics. And I see losses of 15-20 minutes after switching networks.

I think it's better not to use Python code on the client and server sides. It's better to use Golang. I did the same with the Golang library and keepalive and everything works like a swiss clock, even after a network switch. Connections are restored very quickly
